apps/api/src/app/environments-v1/usecases/construct-framework-workflow/construct-framework-workflow.usecase.ts

```diff
@@ -170,22 +178,24 @@ export class ConstructFrameworkWorkflow {
   }
 
   @Instrument()
-  private constructActionStepOptions(staticStep: NotificationStepEntity): Required<Parameters<ActionStep>[2]> {
+  private constructActionStepOptions(
+    staticStep: NotificationStepEntity,
+    fullPayloadForRender: FullPayloadForRender
+  ): Required<Parameters<ActionStep>[2]> {
     return {
-      ...this.constructCommonStepOptions(staticStep),
+      ...this.constructCommonStepOptions(staticStep, fullPayloadForRender),
     };
   }
 
   @Instrument()
-  private constructCommonStepOptions(staticStep: NotificationStepEntity): Required<StepOptions> {
+  private constructCommonStepOptions(
+    staticStep: NotificationStepEntity,
+    fullPayloadForRender: FullPayloadForRender
```

Comment:
we need FullPayloadForRender variables(===frameweork.payload) data in the skip method to execute the skip function.